lib/Client/MultiCurl.php

```diff
@@ -56,7 +56,13 @@ public function __construct($responseFactory, array $options = [])
     {
         parent::__construct($responseFactory, $options);
 
-        if (\PHP_VERSION_ID < 70215 || \PHP_VERSION_ID === 70300 || \PHP_VERSION_ID === 70301 || !(CURL_VERSION_HTTP2 & curl_version()['features'])) {
+        if (
+            \PHP_VERSION_ID < 70215 ||
+            \PHP_VERSION_ID === 70300 ||
+            \PHP_VERSION_ID === 70301 ||
+            \PHP_VERSION_ID >= 80000 ||
```

Comment:
Why disable h2 push on php8?

Reply:
From @chris-doehring :

due to some strange curl behaviour under php 8, the multi curl server push mechanism had to be disabled for php 8. That's something someone needs to check in another pull request.

I was gonna leave the problems with push for you to fix.

Reply:
While I was testing my changes, the `MultiCurlServerPushTest` always failed under php 8, as the `$timeFirstRequest > $timeOtherRequests` assertion was false. I looked into the curl changelog, since it works under php 7.4 but found nothing significant which would make the implementation fail, but either the support in this package, something new of php 8 or in curl is responsible for this. In favor of this PR it was the best in my opinion to deactivate it for php 8 and create a new issue for this so someone can take a deeper dive into it.

Reply:
At least one issue is this weak comparison in the MultiCurl class, since resource comparing changed in php 8, but it wasn't the only issue issue responsible for making the test fail

![image](https://user-images.githubusercontent.com/6341536/96836113-cc215400-1444-11eb-9b44-1dc50407015c.png)

Reply:
Okey. Let's create a new issue about h2 push in php8 and merge this as it is.